### PR TITLE
Add support 'conda debug' calls in build-locally

### PR DESCRIPTION
**Pull request by xhochy**
I manually patch this regularly in `build-locally.py`.

Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

**Comment by xhochy** [comment]
@conda-forge/core This is ready for review, successfully tested this locally with the `zstd-feedstock`.

File: conda_smithy/templates/build_steps.sh.tmpl [hunk 1]
```diff
@@ -39,16 +39,26 @@ if [[ "${HOST_PLATFORM}" != "${BUILD_PLATFORM}" ]]; then
...
+    if [[ "x${BUILD_OUTPUT_ID:-}" != "x" ]]; then
+        EXTRA_CB_OPTIONS="${EXTRA_CB_OPTIONS:-} --output-id ${BUILD_OUTPUT_ID}"
+    fi
+    conda debug "${RECIPE_ROOT}" -m "${CI_SUPPORT}/${CONFIG}.yaml" ${EXTRA_CB_OPTIONS:-}
+    # Drop into an interactive shell
+    /bin/bash
+else
+    conda build "${RECIPE_ROOT}" -m "${CI_SUPPORT}/${CONFIG}.yaml" \
+        --suppress-variables ${EXTRA_CB_OPTIONS:-} \
+        --clobber-file "${CI_SUPPORT}/clobber_${CONFIG}.yaml"
```
Comment (isuruf) [inline conda_smithy/templates/build_steps.sh.tmpl:53]: can you add this to `conda debug` as well?

Comment (xhochy) [reply conda_smithy/templates/build_steps.sh.tmpl:55]: Done and CI passed.

File: conda_smithy/feedstock_content/build-locally.py [hunk 2]
```diff
@@ -16,7 +16,14 @@ def setup_environment(ns):
 
 def run_docker_build(ns):
     script = ".scripts/run_docker_build.sh"
-    subprocess.check_call([script])
+    if ns.debug:
+        env = os.environ.copy()
```
Comment (scopatz) [inline conda_smithy/feedstock_content/build-locally.py:20]: I think it is a little cleaner to move this line above the if-statement and drop the else clause

Comment (isuruf) [reply conda_smithy/feedstock_content/build-locally.py:20]: I think this should be moved to the `setup_environment` function

Comment (scopatz) [reply conda_smithy/feedstock_content/build-locally.py:20]: Yeah that would be fine too

Comment (xhochy) [reply conda_smithy/feedstock_content/build-locally.py]: Done 👍

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**